File: 2156_포도주 시식/wine.js
/*

2017 Dec 19, made by Moon

*/

var fs = require('fs');
var n, c = [];
var input = fs.readFileSync('/dev/stdin').toString().split('\n').map(function(d, i){
	if(i===0){
  	n = parseInt(d);
  }else{
  	c.push(parseInt(d));
  }
});


function main(n, c){
	var cLen, d = [];
  var ix, ixLen;

  c.unshift(0);

  d[0] = 0;
  d[1] = c[1];
  d[2] = d[1] + c[2];

  for(ix = 3, ixLen = n; ix <= ixLen; ix++){
  	if(!d[ix]){
      d[ix] = Math.max(
        d[ix-3] + c[ix-1] + c[ix],
        d[ix-2] + c[ix],
        d[ix-1]
      );
    }
  }

  console.log(d[n]);
};

main(n, c);

File: 2156_포도주 시식/wine.c
/*

2017 Dec 19, made by Jeon

*/


#include <stdio.h>
#define MAX_GLASSES 10001
#define MAX(A,B) A>B?A:B
int main(void)
{
	int n;
	int G[MAX_GLASSES] = { 0 };
	int D[MAX_GLASSES] = { 0 };
	
	scanf("%d", &n);
	for (int i = 1; i <= n; i++)
		scanf("%d", &G[i]);

	D[1] = G[1];
	for (int i = 2; i <= n; i++)
	{
		if (i < 3)
		{
			D[i] = D[i - 1] + G[i];
		}
		else
		{
			int temp = MAX(D[i - 3] + G[i - 1] + G[i], D[i - 1]);
			D[i] = MAX(D[i - 2] + G[i], temp);			
		}
	}

	printf("%d\n", D[n]);
	return 0;
}